src/safety_check.c

```diff
@@ -12,7 +12,6 @@ void safety_check_fail_sized_dealloc(bool current_dealloc) {
 	   " application sized deallocation bugs (source: %s). Suggest building"
 	    "with --enable-debug or address sanitizer for debugging. Abort.\n",
 	    src);
-	abort();
```

Comment:
Should we `assert(config_opt_safety_checks);` at the beginning of this function? Perhaps it should be an or of the two options?

Reply:
Eh; it seems this might be a pain if the number of checks we allow grows. I'm not sure it buys us much.

Reply:
But, is it even correct? I mean, if `config_opt_safety_checks` is turned off, this `safety_check_fail_sized_dealloc()` function can still be called whenever `config_opt_size_checks` is turned on?

Reply:
Oh, I see what you're saying; I missed the assert at the beginning of this function. Yep, you're absolutely right.